model/project_vars.go

```diff
@@ -444,34 +444,53 @@ func (projectVars *ProjectVars) Upsert() (*adb.ChangeInfo, error) {
 
 	projectVars.checkAndRunParameterStoreOp(ctx, func(ref *ProjectRef, isRepoRef bool) {
 		if !ref.ParameterStoreVarsSynced {
-			grip.Error(message.WrapError(FullSyncToParameterStore(ctx, projectVars, ref, isRepoRef), message.Fields{
+			pm, err := FullSyncToParameterStore(ctx, projectVars, ref, isRepoRef)
+			grip.Error(message.WrapError(err, message.Fields{
 				"message":    "could not fully sync project vars into Parameter Store; falling back to using the DB",
 				"op":         "Upsert",
 				"project_id": projectVars.Id,
 				"epic":       "DEVPROD-5552",
 			}))
+			if pm != nil {
+				projectVars.Parameters = *pm
+			}
 		} else {
-			grip.Error(message.WrapError(projectVars.upsertParameterStore(ctx), message.Fields{
+			pm, err := projectVars.upsertParameterStore(ctx)
+			grip.Error(message.WrapError(err, message.Fields{
 				"message":    "could not upsert project vars into Parameter Store; falling back to using the DB",
 				"op":         "Upsert",
 				"project_id": projectVars.Id,
 				"epic":       "DEVPROD-5552",
 			}))
+			if pm != nil {
+				projectVars.Parameters = *pm
+			}
 		}
 	}, "Upsert")
 
+	setUpdate := bson.M{
+		projectVarsMapKey:   projectVars.Vars,
+		privateVarsMapKey:   projectVars.PrivateVars,
+		adminOnlyVarsMapKey: projectVars.AdminOnlyVars,
+	}
+	unsetUpdate := bson.M{}
+	update := bson.M{}
+	if len(projectVars.Parameters) > 0 {
+		setUpdate[projectVarsParametersKey] = projectVars.Parameters
+	} else {
+		unsetUpdate[projectVarsParametersKey] = 1
+	}
+	update["$set"] = setUpdate
+	if len(unsetUpdate) > 0 {
```

Comment:
nit: this conditional can be removed and the next line can be moved up into the above else block.

Reply:
Done.